### PR TITLE
Improve build script for p2p_libp2p

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Improve build script for p2p_libp2p.
- Add coverage.
- Add check for minimum version.
- Improve error messages.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: packages/fetchai/connections/p2p_libp2p/check_dependencies.py [hunk 1]
```diff
@@ -19,19 +19,130 @@
...
+        )
+
+    version_getter_command = [binary_name, *args]
+    stdout = subprocess.check_output(version_getter_command).decode("utf-8")  # nosec
+    version_match = version_regex.search(stdout)
+    if version_match is None:
+        print(
+            f"Warning: cannot parse '{binary_name}' version from command: {version_getter_command}."
+        )
+        return
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_libp2p/check_dependencies.py:117]: the rationale is: if we can't extract the version number, due to weird corner cases, better to don't stop the building, but at least notify the user something didn't go well.

File: packages/fetchai/connections/p2p_libp2p/check_dependencies.py [hunk 2]
```diff
@@ -19,19 +19,130 @@
...
+ERROR_MESSAGE_TEMPLATE_VERSION_TOO_LOW = "The installed version of '{command}' is too low: expected at least {lower_bound}; found {actual_version}."
 
-def check_binary(name: str):
-    """Check a binary is accessible from the terminal."""
-    path = shutil.which(name)
+# for the purposes of this script,
+# a version is a tuple of integers: (major, minor, patch)
+VERSION = Tuple[int, int, int]
+MINIMUM_GO_VERSION: VERSION = (1, 14, 0)
+MINIMUM_GCC_VERSION: VERSION = (7, 5, 0)
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_libp2p/check_dependencies.py:38]: Taken from:
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md#language-and-runtime

However, better ask @lrahmani what is the right minimum requirement for `gcc`.

File: packages/fetchai/connections/p2p_libp2p/check_dependencies.py [hunk 3]
```diff
@@ -19,19 +19,130 @@
...
+ERROR_MESSAGE_TEMPLATE_BINARY_NOT_FOUND = "'{command}' is required by the libp2p connection, but it is not installed, or it is not accessible from the system path."
+ERROR_MESSAGE_TEMPLATE_VERSION_TOO_LOW = "The installed version of '{command}' is too low: expected at least {lower_bound}; found {actual_version}."
 
-def check_binary(name: str):
-    """Check a binary is accessible from the terminal."""
-    path = shutil.which(name)
+# for the purposes of this script,
+# a version is a tuple of integers: (major, minor, patch)
+VERSION = Tuple[int, int, int]
+MINIMUM_GO_VERSION: VERSION = (1, 14, 0)
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_libp2p/check_dependencies.py:37]: taken from the workflow file

File: packages/fetchai/connections/p2p_libp2p/check_dependencies.py [hunk 4]
```diff
@@ -19,19 +19,130 @@
...
+
+    print_ok_message(binary_name, actual_version, version_lower_bound)
+
+
+def main():
+    """The main entrypoint of the script."""
+    check_binary(
+        "go",
+        ["version"],
+        re.compile(r"go version go([0-9]+)\.([0-9]+) "),
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_libp2p/check_dependencies.py:136]: `go` doesn't provide a patch number; we default it to `0`, see how we use the `nth` function.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**